assets/js/modules/tagmanager/components/common/ContainerNameTextField.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * ContainerNameTextField component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_FORMS } from '../../../../googlesitekit/datastore/forms';
+import { TextField, HelperText, Input } from '../../../../material-components';
+import { FORM_SETUP } from '../../datastore/constants';
+import { isUniqueContainerName } from '../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function ContainerNameTextField( { label, containers, formFieldID } ) {
+	const containerName = useSelect( ( select ) => select( CORE_FORMS ).getValue( FORM_SETUP, formFieldID ) );
+
+	const { setValues } = useDispatch( CORE_FORMS );
+	const onChange = useCallback( ( { currentTarget } ) => {
+		setValues( FORM_SETUP, { [ formFieldID ]: currentTarget.value } );
+	}, [ formFieldID ] );
+
+	const helperText = containerName && ! isUniqueContainerName( containerName, containers )
+		? <HelperText persistent>{ __( 'A container with this name already exists.', 'google-site-kit' ) }</HelperText>
+		: undefined;
+
+	let trailingIcon;
+	if ( helperText ) {
+		trailingIcon = (
+			<span className="googlesitekit-text-field-icon--warning">
+				<span className="screen-reader-text">
+					{ __( 'Warning', 'google-site-kit' ) }
+				</span>
```

Comment:
I don't think this screen-reader text is appropriate. The text "Warning" will appear at the end of the textfield's… label, presumably? But it will be quite out of context—as I understand it, this will read: `A container with this name already exists. Warning`. Even if the warning prepended the `helperText`, without a colon it won't be correct: `Warning A container with this name already exists.`. And that won't translate well.

Best to remove the icon's screen-reader text and have it be a visual-only icon. Augmenting the `helperText` to look something like: `<span className="screen-reader-text">Warning: </span>A container with this name already exists.` would be better, but I don't think it's even necessary here.

Reply:
Ok, removed screen reader text.